### PR TITLE
RHEL 9 clones kickstart fixes

### LINKED ISSUE #5513: Kickstart stopped working for RHEL9 related systems
# Problem description

When using a RHEL9 related kickstart profile, the Autoinstallation feature does not seem to work anymore.

It seems that some "install" function has been removed.

## Version of Uyuni Server and Proxy (if used)

```dnf info Uyuni-Server-release
# dnf info Uyuni-Server-release
Letzte Prüfung auf abgelaufene Metadaten: vor 3:39:45 am Sa 28 Mai 2022 08:23:47 CEST.
Installierte Pakete
Name         : Uyuni-Server-release
Version      : 2022.01
Release      : 167.1.uyuni1
Architecture : x86_64
Size         : 1.4 k
Quelle       : Uyuni-Server-release-2022.01-167.1.uyuni1.src.rpm
Repository   : @System
Aus Paketque : @commandline
Summary      : Uyuni Server
Lizenz       : GPL-2.0-or-later
Description  : Uyuni lets you efficiently manage physical, virtual,
             : and cloud-based Linux systems. It provides automated and cost-effective
             : configuration and software management, asset management, and system
             : provisioning.

```
Although product has been updated to release 2022.05. guess it wasn't done properly.

## Details about the issue

Creation of kickstart profile:
1. Sync AlmaLinux 9 baseos/appstream. Tick "create kickstartable tree"
2. Select the Autoinstallation -> distribution to be Red Hat 8 (9 is not available) and verify the data.
3. Create a profile.
Do a network boot and select the new profile.
In addition to the screenshot below, similar errors about "install" being removed show up in the logs flying by.

grafik

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Get AlmaLinux 9 and brothers to start with kickstart.
Also took the opportunity to remove RHEL < 6 related code.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes
- [X] **DONE**

## Test coverage
- No tests: kickstart is not tested for now

- [X] **DONE**

## Links

Fixes #5513
Tracks # **add downstream PR, if any**

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/manager/kickstart/KickstartFormatter.java [hunk 1]
```diff
@@ -754,26 +659,25 @@ else if (this.ksdata.isRhel3() || this.ksdata.isRhel4()) {
...
-                     "/|' -i /etc/sysconfig/rhn/rhn_register" + NEWLINE);
+                    "'s|^(\\s*(noSSLS\\|s)erverURL\\s*=\\s*[^:]+://)[^/]*/|\\${1}" +
+                    up2datehost +
+                    "/|' -i /etc/sysconfig/rhn/up2date" + NEWLINE);
+
+            if (this.ksdata.getVerboseUp2date()) {
+                retval.append("[ -r /etc/sysconfig/rhn/up2date ] && " +
+                        "sed 's/debug=0/debug=1/' -i /etc/sysconfig/rhn/up2date" +
+                        NEWLINE);
+            }
```
Comment (aaannz) [inline java/code/src/com/redhat/rhn/manager/kickstart/KickstartFormatter.java:674]: This is still calling `perl` which is however now removed (or at least not explicitly added) on line 591.

Is this even needed when according to comment this is for rhel2 to 4?

Comment (aaannz) [reply java/code/src/com/redhat/rhn/manager/kickstart/KickstartFormatter.java:674]: I also do not understand what the condition is checking. :D

File: java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java [hunk 2]
```diff
@@ -62,23 +69,14 @@ public boolean isRhel8OrGreater() {
...
-    }
-
-    /**
-     * @return if this installer type is rhel 5 or greater (for rhel6)
-     */
-    public boolean isRhel5OrGreater() {
-        // we need to reverse logic here
-        return (!isRhel2() && !isRhel3() && !isRhel4() && !isFedora() && !isGeneric() &&
-                !isSUSE());
+        return !isFedora() && !isGeneric() && isSUSE();
```
Comment (aaannz) [inline java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java:79]:
```suggestion
        return !isFedora() && !isGeneric() && !isSUSE();
```

File: java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java [hunk 3]
```diff
@@ -102,39 +100,11 @@ public boolean isRhel6() {
...
-    public boolean isRhel2() {
-        return RHEL_21.equals(getLabel());
-    }
-
     /**
      * @return true if the installer type is rhel
      */
     public boolean isRhel() {
-        return isRhel2() || isRhel3() || isRhel4() || isRhel5() || isRhel6() || isRhel7() || isRhel8();
+        return isRhel6() || isRhel7() || isRhel8();
```
Comment (aaannz) [inline java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java:107]: Missing `isRhel9()` and also its implementation.

**Comment by rjmateus** [comment]
I really love this PR's that remove so much code :)

File: java/code/src/com/redhat/rhn/manager/kickstart/KickstartFormatter.java (hunk 1 above)
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/manager/kickstart/KickstartFormatter.java:674]: This if is for traditional client kickstarts on rhel 6 and 7... and those still have perl: I didn't remove it from that code path.

The verbose up2date is just an option the user can click in the UI

File: java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java (hunk 2 above)
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java:79]: yes, found it too while debugging the unit tests failures

File: java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java (hunk 3 above)
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/domain/kickstart/KickstartInstallType.java:107]: thanks. Adding it

**Pull request merged**